### PR TITLE
Conversions: "Convert The Future Into The Present"

### PR DESCRIPTION
**Pull request by mintsoft**
Hello Everyone.

We've been discussing for a while what we want to do around unit conversion. I've been evaluating a number of API's and npm modules as @moollaza knows. I've not found any which cover the same breadth of units that we do currently and that are easily extensible for us to add more later. I feel that given the profile of this IA, it's important for us to be able to customise, fix and maintain it quickly and easily.

I was in conversation with @GuiltyDolphin about what we can do for the future of both Calculator and Conversions. It seems like a natural progression would be to be able to handle queries such as `33 inches + 5 cm in yards`. Extending the grammar for the calculator in https://github.com/duckduckgo/zeroclickinfo-goodies/pull/1927 for units should be relatively trivial, especially if it is handed a list or collection of units from somewhere; however it means we need control over the units and conversions handling etc. 

I think this leads us down a path of modularising both Conversions and Calculator into a collection of Roles (or similar) so we can easily assemble what we need. 

We were also discussing about interactivity of these IA's; we have a couple of choices. One is to re-implement them entirely in JS (relatively trivial for Conversions) however much more difficult for Calculator as grammar parsing isn't very mature in JS. @GuiltyDolphin suggested that there was a conversation around being able to 50-50 it, so that we can build a front-end in JS and do some AJAX off to the backend for processing. This seems like an ideal compromise to me, but I don't believe the IA architecture currently supports such an arrangement.

I've assumed that either way we could do with centralising and tidying this up a bit so I've resurrected an old branch where I brought everything back together and tidied things out in order to kick start the conversation.

This branch:
* Fixes the temperature conversions (includes passing tests from: https://github.com/duckduckgo/zeroclickinfo-goodies/pull/1731)
* Thus fixes https://github.com/duckduckgo/zeroclickinfo-goodies/issues/1413
* Includes https://github.com/duckduckgo/zeroclickinfo-goodies/pull/1992
* Can be easily extended by adding units to the `ratios.yaml` : i.e. resolves: https://github.com/duckduckgo/zeroclickinfo-goodies/issues/828
* Adds support for queries like `How to convert meters to inches` as suggested by @GuiltyDolphin 
* Fixes https://github.com/duckduckgo/zeroclickinfo-goodies/issues/1754
* Fixes (by expanding the existing bodge, until ambiguous units can be resolved, I'd like to tackle that as a separate unit of work) https://github.com/duckduckgo/zeroclickinfo-goodies/issues/1767
* Fixes https://github.com/duckduckgo/zeroclickinfo-goodies/issues/1847 by increasing accuracy of definition rather than rounding the result off sooner
* Fixes https://github.com/duckduckgo/zeroclickinfo-goodies/issues/1729

I would like (if we agree to go down this path) to separate out prefixes from the rest (so all units can be expressed as kiloX or megaX) etc which would simplify things massively and help the modularity.

I'm keen to hear what people think about this, I think we've been a bit stuck-in-the-mud around conversions for too long now.

@duckduckgo/community-leaders @moollaza @jagtalon 

===============
https://duck.co/ia/view/conversions

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by GuiltyDolphin** [comment]
*subscribing again* @mintsoft Very fresh!

**Comment by jagtalon** [comment]
@mintsoft oh my god! This is exciting!

**Comment by jagtalon** [comment]
I'll be testing it out a bunch today. <3

**Comment by javathunderman** [comment]
Exciting!

**Comment by moollaza** [comment]
@mintsoft this looks really promising. I just realized that if the backend continues to return plaintext results, we could potentially leverage the DDG API to build interactivity into the frontend very easily.

Or if absolutely necessary we could likely build a service to handle conversions and return the structured answer as JSON.

I'll get this onto beta so we can give it a test!

**Comment by mintsoft** [comment]
@moollaza Using the DDG API sounds like a good idea actually; at least for an initial version; it'd fit into what @GuiltyDolphin and I were discussing for future revisions

**Comment by mintsoft** [comment]
@moollaza Any idea why the deploy's failed?

**Comment by mintsoft** [comment]
@jagtalon @duckduckgo/community-leaders 

I think this is the most I'd like to do in this PR. I've already discussed with @moollaza about the js side of the things and I'm going to spike something out this weekend hopefully.

There are several issues still here that remain (ambiguous unit resolution is 'fixed' with the bodge around: https://github.com/duckduckgo/zeroclickinfo-goodies/pull/1996/files#diff-e40dae3e0dc357950d79ca5c2e964b3eR74) SI and Binary prefixes are added manually (so only some units support them etc). However I'd like to tackle each of these things (js, ambiguity, prefixes) as separate PRs if this approach is settled upon.

**Comment by gautamkrishnar** [comment]
:heart:

**Comment by mintsoft** [comment]
I've also just gone through all the issues and updated the description with the issues that are solved with this PR

File: lib/DDG/Goodie/Conversions.pm [hunk 1]
```diff
@@ -78,26 +72,20 @@ handle query_lc => sub {
     $_ =~ s/ degrees (celsius|fahrenheit)/ $1/;
```
Comment (moollaza) [inline lib/DDG/Goodie/Conversions.pm]: looks like we don't support `1 degree X to Y` or `X degree(s) (kelvin|rankine) to Y`?

Comment (moollaza) [inline lib/DDG/Goodie/Conversions.pm]: Can fix in next PR -- the live version has this same behaviour anyway...

Comment (mintsoft) [inline lib/DDG/Goodie/Conversions.pm]: Fixed in: https://github.com/duckduckgo/zeroclickinfo-goodies/commit/567b651a3b0ca7d3e48284b115fa8732f9dbdc26

**Comment by moollaza** [comment]
@mintsoft this LGTM! 

Let's get those Travis tests passing again!

**Pull request merged**